platforms/defaults_unix.go

```diff
@@ -38,3 +38,7 @@ func DefaultSpec() specs.Platform {
 func Default() MatchComparer {
 	return Only(DefaultSpec())
 }
+
+func AreWindowsHostAndGuestHyperVCompatible(hostPlatform, guestPlatform specs.Platform) bool {
```

Comment:
it seems like this is a leaky abstraction.  This should be in the Windows matching logic hidden away.